### PR TITLE
expose current iteration item to pipelinerun's annotation

### PR DESCRIPTION
**Pull request by wzhanw**
**Which issue is resolved by this Pull Request:** 
Resolves #
We want to get the current iteration item in the pipeline's task node for condition check and any other purpose.
Any task run created by this pipeline will inherit the annotation list.  In our customer run we can get the item value and to our logic.

**Description of your changes:**
Get the current iteration item and add it to the pipelinerun's annotation list.

### CODE REVIEW

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go [hunk 1]
```diff
@@ -230,7 +233,7 @@ func (c *Reconciler) reconcile(ctx context.Context, run *v1alpha1.Run, status *p
 	}
 
 	// Determine how many iterations of the Task will be done.
-	totalIterations, err := computeIterations(run, pipelineLoopSpec)
+	totalIterations, interationElements, err := computeIterations(run, pipelineLoopSpec)
```
Comment (pugangxa) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:236]: There's many `interationElements`, I guess `interation` should be `iteration`?

Comment (wzhanw) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go]: you are right, I corrected the typo.

**Pull request merged**